### PR TITLE
Nk good to product

### PR DESCRIPTION
**Pull request by NoorAlkattan**
this branch has issue after pull latest dev with crop image and stylesheet 
this branch has changed all goods to products and have add the permission to manage product resources 
my working branch NK-copy. You can validate on this branch

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -32,6 +32,7 @@ gem 'will_paginate-bootstrap4'
 gem "haml-rails", "~> 1.0"
 gem 'trix'
 gem 'shrine'
+gem 'cancancan', '~> 2.0'
```
Comment (MwMaki) [inline Gemfile]: why do you need cancancan for ?

Comment (NoorAlkattan) [reply Gemfile]: give us the permission or ability to manage resources like when i sign in to user table can see control my product and when login to customer can see the item only with anything i can specifiy like show link ..etc.....as i manitioned this branch the crop and style sheets doesn't work properly after merge it with latest dev.

Comment (MwMaki) [reply Gemfile]: But devise gives same capability where you can act as different role like user, admin, customer, business ...ect

Comment (NoorAlkattan) [reply Gemfile]: i don't know of devise can give us ability to modify or to show only inside models

**Pull request merged**